front/app/modules/commercial/representativeness/admin/hooks/useReferenceData.ts

```diff
@@ -0,0 +1,164 @@
+
+export interface RepresentativenessRowPre extends RepresentativenessRowBase {
+  title_multiloc: Multiloc;
+}
+
+export type RepresentativenessData = RepresentativenessRow[];
+
+const getSubscription = (
+  code: TCustomFieldCode | null,
+  fieldId: string,
+  projectId: string | undefined,
+  handleStreamResponse: (usersByField: TStreamResponse) => void
+) => {
+  if (code === null) {
+    const observable = usersByRegFieldStream(
+      { queryParameters: { project: projectId } },
+      fieldId
+    ).observable;
+
+    const subscription = observable.subscribe(handleStreamResponse);
+    return subscription;
+  }
+
+  if (code === 'gender') {
+    const observable = usersByGenderStream({
+      queryParameters: { project: projectId },
+    }).observable;
+
+    const subscription = observable.subscribe(handleStreamResponse);
+    return subscription;
+  }
+
+  if (code === 'domicile') {
+    const observable = usersByDomicileStream({
+      queryParameters: { project: projectId },
+    }).observable;
+
+    const subscription = observable.subscribe(handleStreamResponse);
+    return subscription;
+  }
+
+  return;
+};
+
+function useReferenceData(field: IUserCustomFieldData, projectId?: string) {
+  const [referenceData, setReferenceData] = useState<
+    RepresentativenessRowPre[] | NilOrError
+  >();
+  const [includedUserPercentage, setIncludedUserPercentage] = useState<
+    number | NilOrError
+  >();
+  const [referenceDataUploaded, setReferenceDataUploaded] = useState<
+    boolean | undefined
+  >();
+
+  const code = field.attributes.code;
+  const fieldId = field.id;
+
+  useEffect(() => {
+    const handleStreamResponse = (usersByField: TStreamResponse) => {
+      if (isNilOrError(usersByField)) {
+        setReferenceData(usersByField);
+        setIncludedUserPercentage(usersByField);
+        return;
+      }
+
+      if (!usersByField.series.expected_users) {
+        setReferenceDataUploaded(false);
+        return;
+      }
+
+      const referenceData = toReferenceData(usersByField);
+      const includedUsersPercentage = getIncludedUserPercentage(usersByField);
+      setReferenceData(referenceData);
+      setIncludedUserPercentage(includedUsersPercentage);
+      setReferenceDataUploaded(true);
+    };
+
+    const subscription = getSubscription(
+      code,
+      fieldId,
+      projectId,
+      handleStreamResponse
+    );
+
+    if (subscription) {
+      return () => subscription.unsubscribe();
+    }
+
+    return;
+  }, [code, fieldId, projectId]);
+
+  return {
+    referenceData,
+    includedUserPercentage,
+    referenceDataUploaded,
+  };
+}
+
+export default useReferenceData;
+
+const toReferenceData = (
+  usersByField: TStreamResponse
+): RepresentativenessRowPre[] => {
+  const { users, expected_users } = usersByField.series;
+  const options =
+    'options' in usersByField ? usersByField.options : usersByField.areas;
+  const totalActualUsers = Object.values(users).reduce((acc, v) => v + acc, 0);
```

Comment:
We're re-using `.reduce((acc, v) => v + acc, 0)` a lot. Maybe it makes sense to create a `sum` utility?

```
const sum = (values: number[]) => values.reduce((acc, v) => v + acc, 0);
...
const totalActualUsers = sum(Object.values(users));